mlem/contrib/sagemaker/meta.py

```diff
@@ -0,0 +1,484 @@
+            aws_vars.account,
+            aws_vars.region,
+            image_tag,
+            self.ecr_repository or DEFAULT_ECR_REPOSITORY,
+            aws_vars,
+        )
+        state.image_tag = image_tag
+        meta.update_state(state)
+
+    def remove(self, meta: SagemakerDeployment):
+        with meta.lock_state():
+            state: SagemakerDeployState = meta.get_state()
+            session, aws_vars = self.get_session_and_aws_vars(state.region)
+            if state.model_location is not None:
+                self._delete_model_file(session, state.model_location)
+            if state.endpoint_name is not None:
+
+                client = session.sagemaker_client
+                endpoint_conf = session.sagemaker_client.describe_endpoint(
+                    EndpointName=state.endpoint_name
+                )["EndpointConfigName"]
+
+                model_name = client.describe_endpoint_config(
+                    EndpointConfigName=endpoint_conf
+                )["ProductionVariants"][0]["ModelName"]
+                client.delete_model(ModelName=model_name)
+                client.delete_endpoint(EndpointName=state.endpoint_name)
+                client.delete_endpoint_config(EndpointConfigName=endpoint_conf)
+            if state.image is not None:
+                self._delete_image(meta, state, aws_vars)
+            meta.purge_state()
+
+    def get_status(
+        self, meta: SagemakerDeployment, raise_on_error=True
+    ) -> "DeployStatus":
+        with meta.lock_state():
+            state: SagemakerDeployState = meta.get_state()
+            session = self.get_session(state.region)
+
+            endpoint = session.sagemaker_client.describe_endpoint(
+                EndpointName=state.endpoint_name
+            )
+            status = endpoint["EndpointStatus"]
+            return ENDPOINT_STATUS_MAPPING.get(status, DeployStatus.UNKNOWN)
+
+    def get_session(self, region: str = None) -> sagemaker.Session:
+        return self.get_session_and_aws_vars(region)[0]
+
+    def get_session_and_aws_vars(
+        self, region: str = None
+    ) -> Tuple[sagemaker.Session, AWSVars]:
+        return init_aws_vars(
+            self.profile,
+            self.role,
+            self.bucket,
+            region or self.region,
+            self.account,
+        )
+
+
+def init_aws_vars(
+    profile=None, role=None, bucket=None, region=None, account=None
+):
+    boto_session = boto3.Session(profile_name=profile, region_name=region)
+    sess = sagemaker.Session(boto_session, default_bucket=bucket)
+
+    bucket = (
+        bucket or sess.default_bucket()
+    )  # Replace with your own bucket name if needed
+    region = region or boto_session.region_name
+    config = project_config(project="", section=AWSConfig)
+    role = role or config.ROLE or sagemaker.get_execution_role(sess)
+    account = account or boto_session.client("sts").get_caller_identity().get(
+        "Account"
+    )
+    return sess, AWSVars(
```

Comment:
return a `dict` instead?

Reply:
it's easier to just do `sess, vars = init_aws_vars()` than `res = init_aws_vars(); sess = res["sess"]; vars = res["vars"]`